packages/services/api/src/modules/schema/providers/schema-publisher.ts

```diff
@@ -267,40 +264,53 @@ export class SchemaPublisher {
   async check(input: CheckInput) {
     this.logger.info('Checking schema (input=%o)', lodash.omit(input, ['sdl']));
 
+    const selector = await this.idTranslator.resolveTargetReference({
+      reference: input.target ?? null,
+      onError() {
+        throw new InsufficientPermissionError('schemaCheck:create');
+      },
+    });
```

Comment:
I see we have three terms: `selector`, `target`, `reference`. Would it make more sense if we consolidated the naming some? E.g.

- `Target` / `Resolve Target` / `Fully Qualified Resource Name` (fqrn)
- `Target` / `Resolve Target` / `Fully Qualified Resource Target` (fqrt)
- `Target Expression` / `Resolve Target Expression` / `Target`

The `fqrn`/`fqrt` idea rifts off of the well known fqdm (fully qualified domain name) terminology. And it reflects me wondering if in our system `selector` is/can be generalized to being just an explicit resource target data structure which potentially shows up in other places too.

Reply:
Currenlty, we have the following:
**Target**: A target resource as stored within our database in the `targets` table
**Selector**: An object containing the id breadcrumb of a resource and its parent resources

New in this PR
**Reference**: Input from GraphQL either slug or UUID (see `TargetReferenceInput` input type), that we translate into a `Selector`.

Reply:
@jasonkuhrt Let's move this conversation to another issue to discuss and see if it makes sense!

Reply:
Thanks for the details!